testsuite/features/support/http_client.rb

```diff
@@ -0,0 +1,78 @@
+# Copyright (c) 2022 SUSE LLC.
+# Licensed under the terms of the MIT license.
+
+require 'faraday'
+
+# Wrapper class for HTTP client library (Faraday)
+class HttpClient
+  def initialize(host)
+    puts 'Activating HTTP API'
+    @http_client = Faraday.new('https://' + host)
+  end
+
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/PerceivedComplexity
+  # rubocop:disable Metrics/MethodLength
+  def call(name, params)
+    # Adjust method name
+    name.sub!('auth.', '') if ['auth.login', 'auth.logout'].include?(name)
+    long_name = '/rhn/manager/api/' + name.tr('.', '/')
+    short_name = name.split('.')[-1]
+
+    # Get session cookie from previous calls
+    if params.nil?
+      session_cookie = nil
+    else
+      session_cookie = params[:sessionKey]
+      params.delete(:sessionKey)
+    end
+
+    # Call API
+    if short_name.start_with?('list', 'get', 'is', 'find') || ['logout', 'errata.applicableToChannels'].include?(name)
```

Comment:
Nice :) 
That might be good to go to a constant if it goes longer, but nice for now.

Reply:
I started with an array, but Can told me to do it so.

We have exactly the same logic in the product source code.

If you ask me, it's utterly dirty.